### PR TITLE
Updating README for API additions

### PR DESCRIPTION
**Pull request by prateekcr**
Signed-off-by: Chandrakar, Prateek <prateek.chandrakar@intel.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -201,19 +219,27 @@ There are two ways of running Manufacturer Component Sample:
...
 ```
 $ cd <fido-iot-src>/service/component-samples/manufacturer
-$ mvn -Dmanufacturer_di_port=<manufacturer-server-port> -Dmanufacturer_database_connection_url=<jdbc-url> -Dmanufacturer_database_username=<manufacturer-database-username> -Dmanufacturer_database_password=<manufacturer-database-password> -Dmanufacturer_database_port=<manufacturer-server-database-port> -Dcatalina_home=<path-to-catalina-home> exec:java
+$ mvn -Dmanufacturer_di_port=<manufacturer-server-port> -Dmanufacturer_database_connection_url=<jdbc-url> -Dmanufacturer_database_username=<manufacturer-database-username> -Dmanufacturer_database_password=<manufacturer-database-password> -Dmanufacturer_database_port=<manufacturer-server-database-port> -Dcatalina_home=<path-to-catalina-home> -Dmanufacturer_keystore_password=<manufacturer-keystore-password> -Dmanufacturer_api_user=<manufacturer-api-user> -Dmanufacturer_api_password=<manufacturer-api-password> exec:java
 ```
 
 ### FIDO IoT Manufacturer REST APIs
 
-1. POST /api/v1/assign/<guid>/<id>
+1. POST /api/v1/assign/id=<customer_id>&guid=<guid>
```
Comment (Darshini-Parikh93) [inline README.md:227]: Wouldn't there be a `?` before the parameter values?

File: README.md [hunk 2]
```diff
@@ -201,19 +219,27 @@ There are two ways of running Manufacturer Component Sample:
...
 
    Assigns customer ID to voucher having the input GUID.
 
 2. GET /api/v1/vouchers/<serial_no>
 
    Gets extended voucher with the serial number.
 
+3. POST /api/v1/customers/id=<customer_id>&name=<customer_name>
```
Comment (Darshini-Parikh93) [inline README.md:235]: Same as above.

File: README.md [hunk 3]
```diff
@@ -462,14 +488,38 @@ Some required runtime arguments
...
-  Docker default: 300ms
+  Docker default: 300s
 
 - `owner_to0_rv_blob`
 
   Information shared by RV to device containing network address of prospective owner. Device then uses this information to initiate TO2 protocol.
 
   Docker default: http://localhost:8042?ipaddress=127.0.0.1
 
+- `owner_api_user'
```
Comment (Darshini-Parikh93) [inline README.md:499]: typo: replace the trailing ' by `

File: README.md [hunk 4]
```diff
@@ -493,30 +543,45 @@ There are two ways of running Owner Component Sample:
 
    Returns all voucher available in `TO2_DEVICES` table.
 
-2. GET /api/v1/owner/vouchers/<guid>
+2. GET /api/v1/owner/vouchers/?id=<guid>
 
    Returns the voucher for the specified GUID.
 
-3. POST /api/v1/owner/vouchers/<guid>
+3. POST /api/v1/owner/vouchers/id=<guid>
```
Comment (Darshini-Parikh93) [inline README.md:550]: Add a `?` before passing the parameter id.

Comment (Darshini-Parikh93) [inline README.md:550]: This POST API doesn't use any parameter.

**Comment by prateekcr** [comment]
Fixed as per the comments

File: README.md [hunk 5]
```diff
@@ -201,18 +217,18 @@ There are two ways of running Manufacturer Component Sample:
...
+$ mvn -Dmanufacturer_di_port=<manufacturer-server-port> -Dmanufacturer_database_connection_url=<jdbc-url> -Dmanufacturer_database_username=<manufacturer-database-username> -Dmanufacturer_database_password=<manufacturer-database-password> -Dmanufacturer_database_port=<manufacturer-server-database-port> -Dcatalina_home=<path-to-catalina-home> -Dmanufacturer_keystore_password=<manufacturer-keystore-password> -Dmanufacturer_api_user=<manufacturer-api-user> -Dmanufacturer_api_password=<manufacturer-api-password> exec:java
 ```
 
 ### FIDO IoT Manufacturer REST APIs
 
-1. POST /api/v1/assign/<guid>/<id>
+| Operation                      | Description                        | Path/Query Parameters    | Content Type   |Request Body  | Response Body |
+| ------------------------------:|:----------------------------------:|:------------------------:|:--------------:|-------------:|--------------:|
+| POST /api/v1/assign/?id=<customer_id>&guid=<guid> | Assigns customer ID to voucher having the input GUID. | Query - id: Customer ID, guid = Device GUID | | | |
```
Comment (Darshini-Parikh93) [inline README.md:227]: &guid=<guid>; "\<guid\>" is not rendering in UI. Try adding "\\" before "<" & ">".

File: README.md [hunk 6]
```diff
@@ -307,35 +323,16 @@ There are two ways of running Reseller Component Sample:
...
+| POST /api/v1/resell/vouchers/<serial_number> | Adds voucher to `RT_DEVICES` table against the serial number. | Path - Device Serial Number | application/cbor | Ownership voucher | |
+| DELETE /api/v1/resell/vouchers/<serial_number> | Deletes voucher from `RT_DEVICES` table with the specified serial number. | Path - Device Serial Number | | | |
+| POST /api/v1/resell/customers/?id=<customer_id>&name=<customer_name> | Adds customer keyset to `RT_CUSTOMERS` table. | Query - id: Customer Id, name: Customer Name | text/plain; charset=us-ascii | Customer PEM formatted Public keys | |
+| POST /api/v1/resell/keys/?alias=<alias> | Adds new Reseller keys to the keystore with the given alias | Query - alias: Alias to be used in keystore | | PEM formatted certificate and private key | | 
+| DELETE /api/v1/resell/keys/?alias=<alias> | Deletes the keys corresponding to the input alias from keystore. | Query - alias: Alias to be used in keystore | | | |
```
Comment (Darshini-Parikh93) [inline README.md:333]: Same comment as above for `<alias>`.

File: README.md [hunk 7]
```diff
@@ -307,35 +323,16 @@ There are two ways of running Reseller Component Sample:
...
+| ------------------------------:|:----------------------------------:|:------------------------:|:--------------:|-------------:|--------------:|
+| GET /api/v1/resell/vouchers/<serial_number>?id=<customer_id> | Assigns the customer and returns the extended voucher for the given serial number from `RT_DEVICES` table. | Query - id: Customer Id, Path - Device Serial Number | | | Ownership voucher |
+| POST /api/v1/resell/vouchers/<serial_number> | Adds voucher to `RT_DEVICES` table against the serial number. | Path - Device Serial Number | application/cbor | Ownership voucher | |
+| DELETE /api/v1/resell/vouchers/<serial_number> | Deletes voucher from `RT_DEVICES` table with the specified serial number. | Path - Device Serial Number | | | |
+| POST /api/v1/resell/customers/?id=<customer_id>&name=<customer_name> | Adds customer keyset to `RT_CUSTOMERS` table. | Query - id: Customer Id, name: Customer Name | text/plain; charset=us-ascii | Customer PEM formatted Public keys | |
+| POST /api/v1/resell/keys/?alias=<alias> | Adds new Reseller keys to the keystore with the given alias | Query - alias: Alias to be used in keystore | | PEM formatted certificate and private key | |
```
Comment (Darshini-Parikh93) [inline README.md:332]: Same comment as above for `<alias>`

Comment (Darshini-Parikh93) [inline README.md:332]: knitpick: Query - alias: Alias to be used in keystore -> Query - alias: Alias to be added in keystore

File: README.md (hunk 6 above)
Comment (Darshini-Parikh93) [inline README.md:333]: knitpick: Query - alias: Alias to be used in keystore -> Query - alias: Alias to be removed from the keystore

File: README.md [hunk 8]
```diff
@@ -462,60 +459,69 @@ Some required runtime arguments
...
+   ```
 
-   Adds owner serviceinfo for the GUID in `GUID_OWNERSVI` table.
+### FIDO IoT Owner REST APIs
 
-7. DELETE /api/v1/owner/svi/<guid>
+| Operation                      | Description                        | Path/Query Parameters    | Content Type   |Request Body  | Response Body |
+| ------------------------------:|:----------------------------------:|:------------------------:|:--------------:|-------------:|--------------:|
+| GET /api/v1/owner/vouchers/    | Returns all voucher available in `TO2_DEVICES` table. | | | | Comma-separated list of GUIDs |
+| GET /api/v1/owner/vouchers/?id=<guid> | Returns the voucher for the specified GUID. | Query - id: Device GUID | | | Ownership voucher |
```
Comment (Darshini-Parikh93) [inline README.md:516]: &guid=; "<guid>" is not rendering in UI. Try adding "\" before "<" & ">".

File: README.md [hunk 9]
```diff
@@ -462,60 +459,69 @@ Some required runtime arguments
...
+| POST /api/v1/owner/vouchers/ | Insert voucher against the specified GUID in `TO2_DEVICES` table. | | application/cbor | Content of Ownership voucher in binary format | |
+| DELETE /api/v1/owner/vouchers/?id=<guid> | Deletes voucher of the specified GUID from the `TO2_DEVICES` table. | Query - id: Device GUID | | | |
```
Comment (Darshini-Parikh93) [inline README.md:518]: Same comment as above.

File: README.md [hunk 10]
```diff
@@ -462,60 +459,69 @@ Some required runtime arguments
...
+| POST /api/v1/owner/svivalues/?id=<ServiceInfoId>&isCborEncoded=<boolean> | Adds serviceinfo entry to `OWNER_SERVICEINFO` table. The query parameter 'isCborEncoded' should be 'true' for CBOR encoded binary data that will never be split into smaller chunks while transferring to the device (small in length, to be used for CBOR primitives such as boolean, int, array and map), and it should be 'false' (preferably) for other binary data that could be splitted into smaller chunks and transferred across messages (for example binary values, string). | Query - id: Service info ID, isCborEncoded: Boolean | application/octet-stream or application/cbor | Content of Serviceinfo in binary format. | |
```
Comment (Darshini-Parikh93) [inline README.md:519]: Same comment as above for `<boolean>`.

File: README.md [hunk 11]
```diff
@@ -462,60 +459,69 @@ Some required runtime arguments
...
+| DELETE /api/v1/owner/svivalues/?id=<ServiceInfoId> | Deletes the serviceinfo values from `OWNER_SERVICEINFO` table. | Query - id: Service info ID | | | |
```
Comment (Darshini-Parikh93) [inline README.md:520]: Same comment as above for `<serviceInfoId>`.

File: README.md [hunk 12]
```diff
@@ -462,60 +459,69 @@ Some required runtime arguments
...
+| POST /api/v1/owner/svi/?guid=<guid> | Adds owner serviceinfo for the GUID in `GUID_OWNERSVI` table, that will be transferred to the device in the insertion order. The format is `Entry1,Entry2,Entry3` and so on, where each Entry contains `moduleName:messageName=serviceInfoId`. Here, the 'content' corresponding to the 'serviceInfoId' is transferred to the device. Please see <fido-iot-src>/demo/owner/serviceinfo/sample-svi.csv as an example for the above format, where moduleName is 'sdo_sys' and messageName is either one of 'filedesc', 'write' and 'exec'. The order of each 'Entry' is important as this order decides the sequence in which the Owner will transfer the Service info. | Query - guid: Device GUID | application/text | SVI string | |
```
Comment (Darshini-Parikh93) [inline README.md:521]: Same comment as above for `<guid>`.

File: README.md [hunk 13]
```diff
@@ -462,60 +459,69 @@ Some required runtime arguments
...
+| DELETE /api/v1/owner/svi/?guid=<guid> | Deletes owner serviceinfo for the GUID from the `GUID_OWNERSVI` table. | Query - guid: Device GUID | | | |
```
Comment (Darshini-Parikh93) [inline README.md:522]: Same comment as above for `<guid>`.

File: demo/reseller/README.md [hunk 14]
```diff
@@ -10,4 +10,10 @@ This will copy the required executables and libraries into <fido-iot-src>/demo/r
...
 
-Refer the [Docker Commands](../README.md/#docker-commands) to start the service.
\ No newline at end of file
+Refer the [Docker Commands](../README.md/#docker-commands) to start the service.
+
+***NOTE*** The database file located at <fido-iot-src>/demo/reseller/target/data/reseller.mv.db is not deleted during 'mvn clean'. As as result the database schema and tables are persisted across docker invocations. Please delete the file manually, in case you encounter any error due to the persisted stale data.
+
+# Inserting keys into Reseller keystore
+
+The PKCS12 keystore file <fido-iot-src>/demo/reseller/reseller_keystore.p12 contains the default reseller keys that are imported into the softHSM keystore inside the container, during startup. It contains 3 PrivateKeyEntry with algorithm types: EC-256, EC-384 and RSA-2048, and should continue to hold PrivateKeyEntry with different algorithms. To insert/replace an existing PrivateKeyEntry of any particular algorithm, refer to the section [Inserting Keys into Keystore](../README.md/#inserting-keys-into-keystore) to insert new certificate/private-key pair into <fido-iot-src>/demo/reseller/reseller_keystore.p12.
```
Comment (Darshini-Parikh93) [inline demo/reseller/README.md:19]: `<fido-iot-src>` not getting rendered into the last line. Prefix the angular brackets with "\" wot resolve the issue.

File: demo/reseller/README.md [hunk 15]
```diff
@@ -10,4 +10,10 @@ This will copy the required executables and libraries into <fido-iot-src>/demo/r
 
 # Starting the Reseller service
 
-Refer the [Docker Commands](../README.md/#docker-commands) to start the service.
\ No newline at end of file
+Refer the [Docker Commands](../README.md/#docker-commands) to start the service.
+
+***NOTE*** The database file located at <fido-iot-src>/demo/reseller/target/data/reseller.mv.db is not deleted during 'mvn clean'. As as result the database schema and tables are persisted across docker invocations. Please delete the file manually, in case you encounter any error due to the persisted stale data.
```
Comment (Darshini-Parikh93) [inline demo/reseller/README.md:15]: `<fido-iot-src>` not getting rendered into the last line. Prefix the angular brackets with "\" to resolve the issue.

File: demo/owner/README.md [hunk 16]
```diff
@@ -10,4 +10,10 @@ This will copy the required executables and libraries into <fido-iot-src>/demo/o
 
 # Starting the owner service
 
-Refer the [Docker Commands](../README.md/#docker-commands) to start the service.
\ No newline at end of file
+Refer to the section [Docker Commands](../README.md/#docker-commands) to start the service.
+
+***NOTE*** The database file located at <fido-iot-src>/demo/owner/target/data/ops.mv.db is not deleted during 'mvn clean'. As as result the database schema and tables are persisted across docker invocations. Please delete the file manually, in case you encounter any error due to the persisted stale data.
```
Comment (Darshini-Parikh93) [inline demo/owner/README.md:15]: `<fido-iot-src>` not getting rendered into the last line. Prefix the angular brackets with "\" wot resolve the issue.

File: demo/owner/README.md [hunk 17]
```diff
@@ -10,4 +10,10 @@ This will copy the required executables and libraries into <fido-iot-src>/demo/o
...
+
+# Inserting keys into Owner keystore
+
+The PKCS12 keystore file <fido-iot-src>/demo/owner/owner_keystore.p12 contains the default Owner keys that are imported into the softHSM keystore inside the container, during startup. It contains 3 PrivateKeyEntry with algorithm types: EC-256, EC-384 and RSA-2048, and should continue to hold PrivateKeyEntry with different algorithms. To insert/replace an existing PrivateKeyEntry of any particular algorithm, refer to the section [Inserting Keys into Keystore](../README.md/#inserting-keys-into-keystore) to insert new certificate/private-key pair into <fido-iot-src>/demo/owner/owner_keystore.p12.
```
Comment (Darshini-Parikh93) [inline demo/owner/README.md:19]: `<fido-iot-src>` not getting rendered into the last line. Prefix the angular brackets with "\" to resolve the issue.

File: demo/reseller/README.md (hunk 14 above)
Comment (Darshini-Parikh93) [inline demo/reseller/README.md:19]: The demo component would also be released as a stand-alone binary. We should be including all necessary information without having to refer to other source README files.

File: demo/owner/README.md (hunk 17 above)
Comment (Darshini-Parikh93) [inline demo/owner/README.md:19]: The demo component would also be released as a stand-alone binary. We should be including all necessary information without having to refer to other source README files.

File: demo/manufacturer/README.md [hunk 18]
```diff
@@ -10,4 +10,10 @@ This will copy the required executables and libraries into <fido-iot-src>/demo/m
...
 
-Refer the [Docker Commands](../README.md/#docker-commands) to start the service.
\ No newline at end of file
+Refer the [Docker Commands](../README.md/#docker-commands) to start the service.
+
+***NOTE*** The database file located at <fido-iot-src>/demo/manufacturer/target/data/mfg.mv.db is not deleted during 'mvn clean'. As as result the database schema and tables are persisted across docker invocations. Please delete the file manually, in case you encounter any error due to the persisted stale data.
+
+# Inserting keys into Manufacturer keystore
+
+The PKCS12 keystore file <fido-iot-src>/demo/manufacturer/manufacturer_keystore.p12 contains the default manufacturer keys that are imported into the softHSM keystore inside the container, during startup. It contains 3 PrivateKeyEntry with algorithm types: EC-256, EC-384 and RSA-2048, and should continue to hold PrivateKeyEntry with different algorithms. To insert/replace an existing PrivateKeyEntry of any particular algorithm, refer to the section [Inserting Keys into Keystore](../README.md/#inserting-keys-into-keystore) to insert new certificate/private-key pair into <fido-iot-src>/demo/manufacturer/manufacturer_keystore.p12.
```
Comment (Darshini-Parikh93) [inline demo/manufacturer/README.md:19]: Same 3 comments as mention below for demo/reseller and demo/owner READMEs.

File: README.md [hunk 19]
```diff
@@ -563,3 +575,61 @@ $ mvn -Dfido.iot.pem.dev=<device-PEM-file> exec:java
 The device will be onboarded.
 
 The rendezvous and owner servers must be available during this step.
+
+### Running a demo using Component Samples
+
+1. Start the FIDO IoT Manufacturer Sample as per the steps outlined in [Running the FIDO IoT Manufacturer Sample](#running-the-fido-iot-manufacture-sample).
```
Comment (Darshini-Parikh93) [inline README.md:581]: link not working

File: README.md [hunk 20]
```diff
@@ -563,3 +575,61 @@ $ mvn -Dfido.iot.pem.dev=<device-PEM-file> exec:java
...
+
+2. Complete Device Initialization (DI) by starting the FIDO IoT HTTP Java Device Sample as per the steps outlined in [Running the FIDO IoT HTTP Java Device Sample](#running-the-fido-iot-http-java-device). Delete any existing 'credential.bin' before starting the device.
```
Comment (Darshini-Parikh93) [inline README.md:583]: link not rendering,

**Comment by prateekcr** [comment]
Updated

**Review by Darshini-Parikh93 — APPROVED** [review]

**Review by trbehera — APPROVED** [review]

**Pull request merged**